File: java/src/stressor/utils/AnsiColors.java
package stressor.utils;

public class AnsiColors {
    public static final String ESC = "\u001b";

    // ================= COLOR =================

    public static final String RESET = ESC + "[0m";
    public static final String RED = ESC + "[91m";
    public static final String BRED = ESC + "[101m";
    public static final String GREEN = ESC + "[92m";
    public static final String BGREEN = ESC + "[102m";
    public static final String YELLOW = ESC + "[93m";
    public static final String BYELLOW = ESC + "[103m";
    public static final String BLUE = ESC + "[94m";
    public static final String BBLUE = ESC + "[104m";
    public static final String CYAN = ESC + "[96m";
    public static final String BCYAN = ESC + "[106m";
    public static final String WHITE = ESC + "[97m";
    public static final String BWHITE = ESC + "[107m";
    public static final String GRAY = ESC + "[90m";
    public static final String BGRAY = ESC + "[100m";
    public static final String BLACK = ESC + "[30m";
    public static final String BBLACK = ESC + "[40m";

    // ================= STYLE =================

    public static final String BOLD = ESC + "[1m";
    public static final String ITALIC = ESC + "[3m";
    public static final String UNDERLINE = ESC + "[4m";

    public static String colorize(String str, String... colors) {
        StringBuilder sb = new StringBuilder();
        for (var color : colors) {
            sb.append(color);
        }
        sb.append(str);
        sb.append(RESET);
        return sb.toString();
    }
}

File: rust/src/utils/ansi_colors.rs
pub const RESET: &str = "\x1b[0m";
pub const RED: &str = "\x1b[91m";
pub const BRED: &str = "\x1b[101m";
pub const GREEN: &str = "\x1b[92m";
pub const BGREEN: &str = "\x1b[102m";
pub const YELLOW: &str = "\x1b[93m";
pub const BYELLOW: &str = "\x1b[103m";
pub const BLUE: &str = "\x1b[94m";
pub const BBLUE: &str = "\x1b[104m";
pub const CYAN: &str = "\x1b[96m";
pub const BCYAN: &str = "\x1b[106m";
pub const WHITE: &str = "\x1b[97m";
pub const BWHITE: &str = "\x1b[107m";
pub const GRAY: &str = "\x1b[90m";
pub const BGRAY: &str = "\x1b[100m";
pub const BLACK: &str = "\x1b[30m";
pub const BBLACK: &str = "\x1b[40m";
pub const BOLD: &str = "\x1b[1m";
pub const ITALIC: &str = "\x1b[3m";
pub const UNDERLINE: &str = "\x1b[4m";

#[cfg(feature = "no-ansi")]
#[macro_export]
macro_rules! color {
    ($c:ident) => {
        ""
    };
}

#[cfg(not(feature = "no-ansi"))]
#[macro_export]
macro_rules! color {
    ($c:ident) => {
        $crate::utils::ansi_colors::$c
    };
}

#[cfg(feature = "no-ansi")]
#[macro_export]
macro_rules! colorize {
    ($l:expr; $($c:ident),+) => {
        $l
    };
}

#[cfg(not(feature = "no-ansi"))]
#[macro_export]
macro_rules! colorize {
    ($l:expr; $c:ident $(, $cc:ident)*) => {
        $crate::color!($c).to_owned() $(+ $crate::color!($cc))* + ($l) + $crate::color!(RESET)
    };
}
